### PR TITLE
DM-10645: Known issues in Science Platform Vision Document

### PR DESCRIPTION
**Pull request by ctslater**

### CODE REVIEW

**Comment by gpdf** [comment]
I've committed some additional changes, primarily covering IVOA standards and the standardization of the "Aspect" language, to the same branch from which @ctslater made this PR.

File: LSE-319.tex [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
 \\
 
-These services can be broken down to three different aspects: a web \textbf{Portal}, designed to provide essential data access and visualization services through a simple-to-use website, a \textbf{JupyterLab} environment, that will provide a Jupyter Notebook-like interface enabling next-to-the-data analysis, and an extensive set of \textbf{Web APIs} that the users will be able to use to remotely examine the LSST data set using tools they're already familiar with.
+These services can be broken down to three different ``Aspects:'' a web \textbf{Portal}, designed to provide essential data access and visualization services through a simple-to-use website, a \textbf{Notebook} environment, that will provide a Jupyter Notebook-like interface, based on JupyterLab, enabling next-to-the-data analysis, and an extensive set of \textbf{Web APIs} that the users will be able to use to remotely examine the LSST data set using tools they're already familiar with.
```
Comment (SimonKrughoff) [inline LSE-319.tex]: Seems like the colon should be outside the ticks.

File: LSE-319.tex [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
 \\
 
-This document lays out the high-level vision for the aforementioned aspects and some associated backend services. It is intentionally brief, and meant to generally guide the flow-down of requirements and development product specifications, prioritization, and plans for the Agile development of the relevant elements of the DM system.
-
+This document lays out the high-level vision for the aforementioned Aspects and some associated backend services.
```
Comment (SimonKrughoff) [inline LSE-319.tex:22]: Should "Aspects" be monospace here too?

File: LSE-319.tex [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
+It is intentionally brief, and meant to generally guide the flow-down of requirements and development product specifications, prioritization, and plans for the Agile development of the relevant elements of the DM system.
```
Comment (SimonKrughoff) [inline LSE-319.tex]: development product specifications -- product specification development ?

File: LSE-319.tex (hunk 1 above)
Comment (SimonKrughoff) [reply LSE-319.tex]: Further, the term Aspects is mono spaced here, but not anywhere else in the document.  Is there a reason to do that here, but nowhere else?

File: LSE-319.tex (hunk 2 above)
Comment (SimonKrughoff) [reply LSE-319.tex:22]: See above.

File: backend.tex [hunk 4]
```diff
@@ -1,24 +1,38 @@
...
 
-The user-facing aspects of the LSST Science Platform will be built on top of a number of backend services that can roughly be divided into three categories: \textbf{database services}, \textbf{file services}, and \textbf{batch computing} services (bottom row of Figure~\ref{fig:layeredLSP}). The details of these services are described in the Data Management Design Document (\citeds{LDM-148}) and other associated documents; here, we only provide the high-level guidance as to the capabilities which these services will need to expose to the user (through the three aspects).
+The user-facing Aspects of the LSST Science Platform will be built on top of a number of backend services that can roughly be divided into three categories: \textbf{database services}, \textbf{file services}, and \textbf{batch computing} services (bottom row of Figure~\ref{fig:layeredLSP}).
+The details of these services are described in the Data Management Design Document (\citeds{LDM-148}) and other associated documents; here, we only provide the high-level guidance as to the capabilities which these services will need to expose to the user (through the three Aspects).
```
Comment (SimonKrughoff) [inline backend.tex:4]: Instead of "and other associated documents" I'd prefer something like "and documents referenced therein."  If the documents you are talking about are actually not referenced in LDM-148, they should probably be enumerated here.

File: backend.tex [hunk 5]
```diff
@@ -1,24 +1,38 @@
...
 
 \subsection{Database Services}
 
-Key LSST catalog data products (both for Level 1 and Level 2) will be stored in relational databases and made available for querying by users using the Structured Query Language (SQL) as well as Astronomical Data Query Language (ADQL). These products, and expectations surrounding their schemas, are further described in the \DPDD.
+Key LSST catalogs, both for \textit{Prompt} and \textit{Data Release} data products, will be stored in relational databases and made available for querying by users using
+% the Structured Query Language (SQL) as well as 
+the Astronomical Data Query Language (ADQL), with some restrictions.
+These products, and expectations surrounding their schemas, are further described in the \DPDD.
```
Comment (SimonKrughoff) [inline backend.tex:11]: I've brought this up in various contexts and I'll say it here too. I think it's dangerous to say the DPDD defines a schema.  It really is just an example of what the data products could look like, but we know for a fact the final catalogs won't share a schema with the DPDD verbatim.

File: backend.tex [hunk 6]
```diff
@@ -1,24 +1,38 @@
...
+Besides serving the LSST catalogs, LSST databases will also provide a per-user database space allocation.
+Within this allocation, end-users (including groups) will be able to store selected or transformed subsets of the LSST dataset, or upload related datasets for joining to the LSST dataset.
+The size of this allocation is determined by the \SRD requirement to provide 10\% of total LSST computing and storage resources to LSST users.
+All users will have a small initial allocation of space, with the possibility of applying to a Resource Allocation Committee for additional quota.
 
 \subsection{File Services}
 
-LSST Science Platform will also provide a per-user file space allocation. End-users (including groups) may use this allocation to upload code, store selected or transformed subsets of the LSST dataset (e.g., images), and in general keep files needed to support their data analysis work. Note that some of this space may be provided in form of an object store, rather than a file system with POSIX-like semantics.
+LSST Science Platform will also provide a per-user file space allocation.
```
Comment (SimonKrughoff) [inline backend.tex]: "The LSST Science Platform" ?

File: backend.tex [hunk 7]
```diff
@@ -1,24 +1,38 @@
...
+Analysis performed through the Portal, Notebook, and Web API Aspects will be served by a shared computing cluster.
+This cluster will be managed by a workload management system that ensures resources are allocated to individual users or groups based on pre-determined operational policies.
+The size of the batch computing resource is determined by the \SRD requirement to provide 10\% of total LSST computing and storage resources to LSST users.
+Again, all users will have a small initial allocation of batch CPU time, with the possibility of applying to a Resource Allocation Committee for additional quota.
 
-The users will be able to launch jobs on the batch computing cluster primarily utilizing the APIs exposed through the JupyterLab and Web API aspects of the LSST Science Platform. Some functionality exposed through the Portal may potentially utilize the batch computing cluster as well.
+The users will be able to launch jobs on the batch computing cluster primarily utilizing the APIs exposed through the Notebook and Web API Aspects of the LSST Science Platform.
```
Comment (SimonKrughoff) [inline backend.tex:34]: I thought the baseline was that the batch computing would be launched from the terminal prompt, so not from the notebook at all.  I'm even more surprised the Web API Aspect will have the ability to launch batch jobs for users.

File: intro.tex [hunk 8]
```diff
@@ -41,41 +47,50 @@ \subsection{LSST Science Platform Overview}
 set of services that will be offered to LSST users.
 
 The platform exposes the LSST data
-and services to the user through three primary user-facing ``\textit{aspects}'' -- the web \textbf{Portal},
-the \textbf{JupyterLab} analysis environment, and a machine-accessible \textbf{Web API} interface. These aspects provide three different ways to access the data sets and analysis services provided in the LSST Data Access Centers (Figure~\ref{fig:layeredLSP}).
+and services to the user through three primary user-facing ``\textit{Aspects}'' --- the web \textbf{Portal},
```
Comment (SimonKrughoff) [inline intro.tex]: Here "Aspects" is monospace and italics.  I think we need a quick pass through to make it self consistent.

File: intro.tex [hunk 9]
```diff
@@ -41,41 +47,50 @@ \subsection{LSST Science Platform Overview}
...
 
-The second, \textbf{JupyterLab}, aspect will provide a Jupyter Notebook-like interface, and
-is geared towards enabling next-to-the-data analysis. The user experience will
+The \textbf{Notebook Aspect} will provide a Jupyter Notebook-like interface,
+based on JupyterLab, and is geared towards enabling next-to-the-data analysis.
+The user experience will
 be nearly identical to working with Jupyter notebooks locally, except that computation
-and analysis will occur with resources provided at the LSST Data Access Center.  This is an
-implementation of the “bringing computation to the data” paradigm: rather
+and analysis will occur with resources provided at the LSST Data Access Center.
```
Comment (SimonKrughoff) [inline intro.tex:64]: I usually refer to the LSP running at the LDF.  If that's wrong, I should change the words I use, if not, I think being consistent here would be helpful.

File: intro.tex [hunk 10]
```diff
@@ -41,41 +47,50 @@ \subsection{LSST Science Platform Overview}
...
+This interface will open the
+possibility for remote access and analysis of the LSST data set using
+applications that the users are already comfortable with (e.g., TOPCAT or libraries like Astropy).
+Furthermore, the offered APIs will allow for federation
 with other astronomical archives, bringing added value to the LSST dataset.
 We describe it in more detail in \secref{sec:apis}.
 
-Enabling these user-facing aspects is a set of backend services. The Data Releases will be organized as catalogs kept in relational database management systems, as well as repositories of files. The alert distribution system will facilitate the distribution of Alert Streams to community brokers and end-users (see the \DPDD for details). These services will be complemented by additional User Database, File Storage, and Batch Computing services, as well as pre-installed Software Tools suite. They will provide the computational power, data storage, and analytics capabilities needed to enable LSST data analysis as well as the creation and federation of Level 3 products. We further describe these in \secref{sec:backend}.
+Enabling these user-facing Aspects is a set of backend services.
```
Comment (SimonKrughoff) [inline intro.tex]: This sentence is a bit clunky:
These user-facing Aspects are enabled by a set of backend services.
?

File: methodology.tex [hunk 11]
```diff
@@ -4,42 +4,56 @@ \subsection{Iterative Development Leveraging Existing Technologies }
 
 The services constructed for the LSST Science Platform will be developed following the iterative Agile methodology. While most of LSST software development follows this approach, adopting it is especially advantageous for user-facing services. There, iterative development and nearly continuous stakeholder feedback can provide guidance as to the details of features to be implemented, the continued validity of the approach taken, and the expected focus of intermediate milestones.
 
-The development of the Portal, JupyterLab, as well as Web API aspects will start from significant existing code bases and prior art. This is a deliberate approach designed to minimize technological risk and leverage end-user familiarity with these interfaces. The latter also reduces the barrier to user adoption of the products eventually delivered for LSST.
+The development of the Portal, Notebook, and Web API Aspects will start from significant existing code bases and prior art.
+This is a deliberate approach designed to minimize technological risk and leverage end-user familiarity with these interfaces.
```
Comment (SimonKrughoff) [inline methodology.tex]: I think I would have chosen "technical risk" over "technological risk".

File: methodology.tex [hunk 12]
```diff
@@ -4,42 +4,56 @@ \subsection{Iterative Development Leveraging Existing Technologies }
...
+The latter also reduces the barrier to user adoption of the products eventually delivered for LSST.
 
-The \textbf{Portal} is based on existing, production quality, archive portal interface developed at IRSA/IPAC -- the \emph{Firefly} toolkit. The primary challenge is integrating the existing Firefly code, and updating the user experience to conform to anticipated user expectations (e.g., supporting all-sky maps and pan/zoom/click-type exploration). Consistent with the general philosophy, DM should look at achieving the necessary upgrades by re-using existing well-known libraries and tools (e.g. Aladin Lite).
+The \textbf{Portal Aspect} is based on the existing, production quality, archive portal interface developed at IRSA/IPAC --- the \emph{Firefly} toolkit.
```
Comment (SimonKrughoff) [inline methodology.tex:11]: This is the first instance of the use of `\emph`.  `\textit` is used elsewhere.  I think I'd prefer `\emph` everywhere, but that seems out of scope for this ticket.

File: methodology.tex [hunk 13]
```diff
@@ -4,42 +4,56 @@ \subsection{Iterative Development Leveraging Existing Technologies }
...
 
-Finally, the \textbf{Web API} aspect is envisioned as implementing existing, widely-adopted, community protocols (e.g. such as those from Virtual Observatory suite of protocols and standards). Similarly to other aspects, it will benefit from leveraging existing codes and libraries wherever appropriate.
+Finally, the \textbf{Web API Aspect} is envisioned as implementing existing, widely-adopted, community protocols (in particular, ones from the Virtual Observatory suite of protocols and standards).
+Similarly to the other Aspects, it will benefit from leveraging existing codes and libraries wherever appropriate.
+We view providing a robust TAP (including ObsTAP) service as particularly critical, both for external and internal use.
+Other IVOA standards that we expect to support include SCS for simple catalog searches, SIAv2 for image searches, SODA for image cutouts and mosaics, and VOSpace (in addition to WebDAV) for access to user files.
```
Comment (SimonKrughoff) [inline methodology.tex:21]: I know it can be a pain to implement, but I find it infinitely useful when links to the standard document is included with the mention of these protocols.

File: user.tex [hunk 14]
```diff
@@ -38,52 +38,53 @@ \subsection{Web Portal\label{sec:portal}}
...
-quality assessment, and science validation of the as-built system. It will be the primary
-method of performing interactive analysis of acquired data (e.g. adjusting and executing
+The Notebook Aspect of the science platform will play a key role in commissioning,
+quality assessment, and science validation of the as-built system.
+It will be the primary
+method of performing interactive analysis of acquired data (e.g., adjusting and executing
 prepared notebooks driving commissioning tasks), as well as commanding
-the batch resources to execute larger processing tasks. Due to this, we expect the
-JupyterLab aspect to reach maturity earlier than the others, and certainly in time for
+the batch resources to execute larger processing tasks.
```
Comment (SimonKrughoff) [inline user.tex]: I'll say again here.  I don't believe the baseline is to launch batch jobs from the notebook.  I think we should keep expectations in line with what we are planning on delivering.

Comment (SimonKrughoff) [reply user.tex]: If we say commanding batch resources through the terminal functionality of the JupyterLab environment, I would be much happier.

**Review by SimonKrughoff — APPROVED** [review]
These are mainly editorial/typographic.  My only real worry is that it seems like we are promising more access to batch compute than I understood was on the roadmap.  I might well be wrong.

File: LSE-319.tex (hunk 1 above)
Comment (gpdf) [reply LSE-319.tex]: 1) You are correct about the colon; I forgot that colons and semicolons are treated the opposite way from commas.  (And neither are "logical" quoting in American practice - they go inside, or outside, regardless of whether they are part of what's being quoted.)  Amusingly, in the previous paragraph, a comma is misplaced in this regard, in the opposite direction.

2) I think those are quotation marks, not monospace.  (LaTeX markup vs. MarkDown?)

File: LSE-319.tex (hunk 2 above)
Comment (gpdf) [reply LSE-319.tex:22]: "Aspects" is quoted when originally defined, at the beginning of the document, and then not again.  It's not monospace.

File: LSE-319.tex (hunk 3 above)
Comment (gpdf) [reply LSE-319.tex]: Development "of" product specifications is what I think was originally intended, for parallel construction, and what I'll do in the absence of further comment.

File: LSE-319.tex (hunk 1 above)
Comment (SimonKrughoff) [reply LSE-319.tex]: Argh.  Very sorry.  I was reading this as rST.

File: backend.tex (hunk 4 above)
Comment (gpdf) [reply backend.tex:4]: I'm not sure what the full set of documents that would be appropriate to mention actually is.  Most likely the original intent was to just indirect through LDM-148 in order to avoid making a project-level change-controlled document dependent on a whole list of DM-level documents.  I think that was probably a good idea.

File: backend.tex (hunk 5 above)
Comment (gpdf) [reply backend.tex:11]: I thought that "expectations surrounding their schemas" was a way to finesse that.  The DPDD does _constrain_ the schemas even though it doesn't _specify_ them.

File: intro.tex (hunk 10 above)
Comment (gpdf) [reply intro.tex]: "The user-facing Aspects all depend on a common set of back-end services, which facilitate the LSP's operation as an integrated whole."?

File: methodology.tex (hunk 11 above)
Comment (gpdf) [reply methodology.tex]: Will do.

File: methodology.tex (hunk 13 above)
Comment (gpdf) [reply methodology.tex:21]: That'll take a little time; I'll do that during the LCR period.  There are other document references that I think we need to add as well.

File: user.tex (hunk 14 above)
Comment (gpdf) [reply user.tex]: Sounds good to me.

File: methodology.tex (hunk 12 above)
Comment (gpdf) [reply methodology.tex:11]: I'll look at the whole document again and harmonize.

File: backend.tex (hunk 5 above)
Comment (leannep) [reply backend.tex:11]: Simon's point is a good one but I tend to agree wit Gregory that  that 'expectations surrounding their schemas' covers it.

File: backend.tex (hunk 4 above)
Comment (gpdf) [reply backend.tex:4]: So, to be clear, I'm not doing anything about this right now and will leave this "conversation unresolved" in the PR.

File: backend.tex (hunk 7 above)
Comment (gpdf) [reply backend.tex:34]: That was originally intended.  It's a "Vision".  I think this is unlikely to happen in "Version 1" of the system.  This probably should be reworded.

File: methodology.tex (hunk 12 above)
Comment (gpdf) [reply methodology.tex:11]: Resolved all in favor of `\emph` - there were comparable numbers of each.

File: backend.tex (hunk 5 above)
Comment (gpdf) [reply backend.tex:11]: Leaving this unchanged.

**Comment by gpdf** [comment]
[LCR-1729](https://project.lsst.org/groups/ccb/node/3107) has been raised for the approval of this document.
